presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/security/FileBasedAccessControl.java

```diff
@@ -333,14 +340,44 @@ private boolean checkTablePermission(ConnectorSecurityContext context, SchemaTab
         return false;
     }
 
+    private boolean checkTableAnyPermission(ConnectorSecurityContext context, SchemaTableName tableName)
+    {
+        if (INFORMATION_SCHEMA_NAME.equals(tableName.getSchemaName())) {
+            return true;
+        }
+
+        for (TableAccessControlRule rule : tableRules) {
+            Optional<Set<TablePrivilege>> tablePrivileges = rule.match(context.getIdentity().getUser(), tableName);
+            if (tablePrivileges.isPresent()) {
+                if (tablePrivileges.get().isEmpty()) {
+                    return false;
+                }
+                return true;
+            }
+        }
+        return false;
+    }
+
     private boolean isSchemaOwner(ConnectorSecurityContext context, String schemaName)
     {
         for (SchemaAccessControlRule rule : schemaRules) {
-            Optional<Boolean> owner = rule.match(context.getIdentity().getUser(), schemaName);
-            if (owner.isPresent()) {
-                return owner.get();
+            Optional<Set<SchemaPrivilege>> privileges = rule.match(context.getIdentity().getUser(), schemaName);
+            if (privileges.isPresent()) {
+                return privileges.get().contains(SchemaPrivilege.OWNERSHIP);
+            }
+        }
+        return false;
+    }
+
+    private boolean isSchemaVisibleToUser(ConnectorSecurityContext context, String schemaName)
+    {
+        for (SchemaAccessControlRule rule : schemaRules) {
+            Optional<Set<SchemaPrivilege>> privileges = rule.match(context.getIdentity().getUser(), schemaName);
+            if (privileges.isPresent()) {
+                return privileges.get().contains(SchemaPrivilege.OWNERSHIP) || privileges.get().contains(VISIBLE);
```

Comment:
Having VISIBLE privilege makes possible that user could see (have any access) to a table in given schema, but schema could be invisible.

Instead of adding new privilege I would iterate over all table privileges and make schema visible if there is any matching table privilege.

Reply:
> Having VISIBLE privilege makes possible that user could see (have any access) to a table in given schema, but schema could be invisible.
> 
> Instead of adding new privilege I would iterate over all table privileges and make schema visible if there is any matching table privilege.
@kokosing 
I don't think ` VISIBLE ` privilege is equivalent to access to table
Access to table is still guaranteed by ` checkTablePermission ` and ` hasAnyTablePermission `